src/Core/Repository/AbstractMultiShopObjectModelRepository.php

```diff
@@ -173,4 +174,42 @@ protected function checkShopAssociation(int $id, string $objectModelClassName, S
             ));
         }
     }
+
+    /**
+     * @param ObjectModel $objectModel
+     * @param ShopId[]|int[] $shopIds
+     * @param string $exceptionClass
+     * @param int $errorCode
+     *
+     * @throws CoreException
+     */
+    protected function deleteObjectModelFromShops(ObjectModel $objectModel, array $shopIds, string $exceptionClass, int $errorCode = 0): void
```

Comment:
are you sure its a good idea to accept 2 different types of $shopIds? I get it why, but it just seem to be inapropriate :smile: )
Also side note - we will get some conflicts on this file. I changed array $shopIds to only support ShopId classes in my pr https://github.com/PrestaShop/PrestaShop/pull/28395/files#diff-0fa51bc40a6381936d12a96d9c2061691dd5b584913836e6d13ee2db7a9a0335R89